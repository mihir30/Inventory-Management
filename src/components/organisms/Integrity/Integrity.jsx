import React, { useEffect, useState } from 'react'
import IntegritySection from '../../molecules/IntegritySection/IntegritySection'
import NavigationBar from '../../molecules/NavigationBar/NavigationBar'
import { getData } from '../../utils/common.utils';

const Integrity = () => {
    const [unitsList, setUnitsList] = useState([]);
    const [equipmentList,setEquipmentList] = useState([]);
    const url = window.location.href;
    useEffect(() => {
        const fetchData = async () => {
            try {
                const units = await getData('http://localhost:8000/units');
                const equipments = await getData('http://localhost:8000/equipments')
                setUnitsList(units);
                setEquipmentList(equipments);

            } catch (error) {
                console.log("error", error.message);
            }
        };
        fetchData();
    }, [url]);

    let archivedEquipments = [];
    let countArchivedStored = 0;
    let countArchivedEl = 0;
    let countArray = [];

    for(let i = 0; i < equipmentList.length; i++){
        if(equipmentList[i].archive === true){
            archivedEquipments.push(equipmentList[i]);
        }
    }
    
    for(let i = 0; i<archivedEquipments.length; i++){
        for(let j = 0; j<unitsList.length; j++){
            if(archivedEquipments[i].name === unitsList[j].type){
                countArchivedStored+=1;
            }
        }
    }
    const userExists = (name) => {
        return equipmentList.some((el) => {
            for(let i = 0; i< el.parts.length; i++){
                if(el.parts[i].partName === name){
                    countArchivedEl+=1;
                }
            }

            countArray.push(countArchivedEl);

            return countArchivedEl;
          
        }); 
    }
    
    
    for(let i = 0; i<archivedEquipments.length; i++){
        userExists(archivedEquipments[i].name);
    }

    const countArchivedParts= countArray.reduce(add, 0);

    function add(accumulator, a) {
      return accumulator + a;
    }
    
    return (
        <div><NavigationBar></NavigationBar><IntegritySection archivedPartsValue={countArchivedParts} archivedStoredValue={countArchivedStored}></IntegritySection></div>
    )
}

export default Integrity