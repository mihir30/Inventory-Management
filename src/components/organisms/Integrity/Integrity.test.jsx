import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ReactDom from "react-dom";
import Integrity from './Integrity';

describe("Integrity Component", () => {
    it('renders the component properly', () => {
        const divEl = document.createElement("div");
        ReactDom.render(<BrowserRouter><Integrity></Integrity></BrowserRouter>, divEl); 
        
    });

    it('should have the right integrity component snapshot', () => {
        const tree = renderer.create().toJSON();
        expect(tree).toMatchSnapshot();
    })

});