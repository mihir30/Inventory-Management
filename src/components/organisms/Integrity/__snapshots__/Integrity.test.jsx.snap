// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integrity Component should have the right integrity component snapshot 1`] = `null`;
